Build error: error TS2304: Cannot find name 'inspect'.

My project has a dev dependency on `homebridge`. The latest beta versions of `homebridge` have dependencies on Matter, so note that this is an indirect dependency. I am not directly importing any Matter packages. When I try to upgrade `homebridge` to one of those versions, I get the following build error:

```
Error: node_modules/@matter/model/dist/esm/models/CrossReference.d.ts(14,6): error TS2304: Cannot find name 'inspect'.
Error: node_modules/@matter/model/dist/esm/models/Model.d.ts(255,6): error TS2304: Cannot find name 'inspect'.
```

The build fails both locally and in github, with Node versions 18, 20, 22, 24.

A quick search brought this up:
> Error TS2304: Cannot find name '{0}' is a compile-time error that occurs when TypeScript cannot recognize a name (like a variable, type, class, or function) in your code. It’s a type of name-resolution error, meaning that TypeScript is unable to find the definition or declaration of the name you’re trying to reference.

My build script:

```
  "scripts": {
    "build": "rimraf ./dist && tsc"
  },
```

I updated my `tsconfig.json` as per the [Typescript notes](https://github.com/matter-js/matter.js/blob/main/packages/main/README.md#typescript-note):

```
{
  "compilerOptions": {
    "target": "ES2022",
    "lib": [
      "DOM",
      "ES2022"
    ],
    "rootDir": "src",
    "module": "node16",
    "moduleResolution": "node16",
    "strict": true,
    "declaration": true,
    "outDir": "dist",
    "sourceMap": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "noImplicitAny": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "resolveJsonModule": true
  },
  "include": [
    "eslint.config.js",
    "homebridge-ui",
    "src"
  ]
}
```

--- a/packages/general/src/MatterError.ts
+++ b/packages/general/src/MatterError.ts
@@ -4,11 +4,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { NodeJsStyleInspectable } from "#log/NodeJsStyleInspectable.js";
 import type { MaybePromise } from "#util/Promises.js";
 import { decamelize } from "#util/String.js";
 import { errorOf } from "./util/Error.js";
-
-const inspect = Symbol.for("nodejs.util.inspect.custom");
 
 const codes = new WeakMap<{}, string>();
 
@@ -80,28 +79,6 @@ export class MatterError extends Error {
         }
 
         return id;
-    }
-
-    /**
-     * Node.js-style object inspection.
-     *
-     * Node's default inspection only prevents two levels of depth which may hide critical information.  It's also
-     * considerably more verbose than native matter.js formatting.  We therefore offer this custom implementation.
-     *
-     * Note that this conforms to Node's API but is not dependent on Node.
-     */
-    [inspect](depth: number, inspectionOptions?: { colors?: boolean }) {
-        const formatterFor = MatterError.formatterFor;
-        if (typeof formatterFor !== "function") {
-            return this;
-        }
-
-        const format = formatterFor(inspectionOptions?.colors ? "ansi" : "plain");
-        if (typeof format !== "function") {
-            return this;
-        }
-
-        return format(this, depth);
     }
 
     /**
@@ -192,9 +169,6 @@ export class MatterAggregateError extends AggregateError {
     constructor(causes: Iterable<unknown>, message?: string) {
         causes = [...causes].map(errorOf);
         super(causes, message);
-
-        Object.defineProperty(MatterAggregateError.prototype, inspect, { enumerable: false });
-        Object.defineProperty(MatterAggregateError.prototype, "format", { enumerable: false });
     }
 
     /**
@@ -246,12 +220,9 @@ export class MatterAggregateError extends AggregateError {
         }
         return (results as PromiseFulfilledResult<T>[]).map(result => result.value);
     }
-}
 
-Object.assign(MatterAggregateError, {
-    [inspect]: MatterError.prototype[inspect],
-    format: MatterError.prototype.format,
-});
+    format = MatterError.prototype.format;
+}
 
 /**
  * It's never reasonable to fail to present error information so we include this rudimentary fallback error formatter.
@@ -348,3 +319,34 @@ export class AbortedError extends CanceledError {
         return super.accept(cause);
     }
 }
+
+/**
+ * Node.js-style object inspection.
+ *
+ * Node's default inspection only prevents two levels of depth which may hide critical information.  It's also
+ * considerably more verbose than native matter.js formatting.  We therefore offer this custom implementation.
+ *
+ * Note that this conforms to Node's API but is not dependent on Node.
+ */
+function inspector(this: MatterError, depth: number, options: NodeJsStyleInspectable.Options) {
+    const formatterFor = MatterError.formatterFor;
+    if (typeof formatterFor !== "function") {
+        return this;
+    }
+
+    const format = formatterFor(options.colors ? "ansi" : "plain");
+    if (typeof format !== "function") {
+        return this;
+    }
+
+    return format(this, depth);
+}
+
+NodeJsStyleInspectable(MatterError.prototype, inspector);
+NodeJsStyleInspectable(MatterAggregateError.prototype, inspector);
+Object.defineProperty(MatterAggregateError.prototype, "format", {
+    value: MatterError.prototype.format,
+    configurable: true,
+    writable: true,
+    enumerable: false,
+});

--- a/packages/general/src/log/NodeJsStyleInspectable.ts
+++ b/packages/general/src/log/NodeJsStyleInspectable.ts
@@ -1,0 +1,37 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const inspect = Symbol.for("nodejs.util.inspect.custom");
+
+/**
+ * Customize object inspection using a Node.js-style inspector function.
+ */
+export function NodeJsStyleInspectable<T extends object>(target: T, inspector: NodeJsStyleInspectable.Inspector<T>) {
+    const invokeInspector: NodeJsStyleInspectable.Inspector<T> = function (depth, options, inspect) {
+        depth ??= 0;
+        options ??= {};
+        inspect ??= target => `${target}`;
+
+        inspector.call(this, depth, options, inspect);
+    };
+
+    Object.defineProperty(target, inspect, {
+        value: invokeInspector,
+        enumerable: false,
+        configurable: true,
+        writable: true,
+    });
+}
+
+export namespace NodeJsStyleInspectable {
+    export interface Inspector<T extends object> {
+        (this: T, depth: number, options: Options, inspect: (value: unknown, options?: Options) => string): unknown;
+    }
+
+    export interface Options extends Record<string, unknown> {
+        color?: boolean;
+    }
+}

--- a/packages/general/src/log/index.ts
+++ b/packages/general/src/log/index.ts
@@ -11,3 +11,4 @@ export * from "./LogDestination.js";
 export * from "./LogFormat.js";
 export * from "./Logger.js";
 export * from "./LogLevel.js";
+export * from "./NodeJsStyleInspectable.js";

--- a/packages/general/src/time/TimeUnit.ts
+++ b/packages/general/src/time/TimeUnit.ts
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { NodeJsStyleInspectable } from "#log/NodeJsStyleInspectable.js";
 import { Duration } from "./Duration.js";
 
 /**
@@ -93,8 +94,9 @@ export function TimeUnit<T = {}>(kind: TimeUnit.Kind, abbrev: string, one: numbe
         floor,
         round,
         toString: kindOf,
-        [Symbol.for("nodejs.util.inspect.custom")]: kindOf,
     });
+
+    NodeJsStyleInspectable(unit, kindOf);
 
     return unit;
 }

--- a/packages/model/src/models/CrossReference.ts
+++ b/packages/model/src/models/CrossReference.ts
@@ -5,8 +5,7 @@
  */
 
 import { Specification } from "#common/Specification.js";
-
-const inspect = Symbol.for("nodejs.util.inspect.custom");
+import { NodeJsStyleInspectable } from "#general";
 
 export class CrossReference implements Specification.CrossReference {
     document: Specification;
@@ -39,11 +38,11 @@ export class CrossReference implements Specification.CrossReference {
 
         return (this.instances[key] = new CrossReference(xref));
     }
-
-    [inspect](_depth: any, options: any, inspect: any) {
-        return inspect ? inspect(this.toString(), options) : this.toString();
-    }
 }
+
+NodeJsStyleInspectable(CrossReference.prototype, function (_depth, options, inspect) {
+    return inspect(this.toString(), options);
+});
 
 export namespace CrossReference {
     export type Definition = Specification.CrossReference | `${string}§${string}`;

--- a/packages/model/src/models/Model.ts
+++ b/packages/model/src/models/Model.ts
@@ -4,15 +4,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { camelize, decamelize, ImplementationError } from "#general";
+import { camelize, decamelize, ImplementationError, NodeJsStyleInspectable } from "#general";
 import { DefinitionError, ElementTag, Metatype, Specification } from "../common/index.js";
 import { AnyElement, BaseElement } from "../elements/index.js";
 import { ModelTraversal } from "../logic/ModelTraversal.js";
 import { Children, InternalChildren } from "./Children.js";
 import type { MatterModel } from "./MatterModel.js";
 import { Resource, ResourceBundle } from "./Resource.js";
-
-const inspect = Symbol.for("nodejs.util.inspect.custom");
 
 /**
  * Thrown when model API is used incorrectly.
@@ -726,30 +724,30 @@ export abstract class Model<E extends BaseElement = BaseElement, C extends Model
     get hasLocalResource() {
         return !!(this.#resource || (this.#root?.resources || ResourceBundle.default).get(this));
     }
-
-    [inspect](_depth: any, options: any, inspect: any) {
-        const json = this.valueOf() as Record<string, unknown>;
-        const props = {
-            name: json.name,
-        } as Record<string, unknown>;
-        if (json.id !== undefined) {
-            props.id = json.id;
-        }
-        for (const key in json) {
-            if (key === "id" || key === "name" || key === "tag") {
-                continue;
-            }
-            props[key] = json[key];
-        }
-        if (this.#children !== undefined && this.#children.length) {
-            props.children = this.#children.length;
-        }
-        if (!inspect) {
-            inspect = (value: unknown) => `${value}`;
-        }
-        return `${inspect(props, options)}`.replace(/^\{/, `${decamelize(this.tag)} {`);
-    }
 }
+
+NodeJsStyleInspectable(Model.prototype, function (_depth, options, inspect) {
+    const json = this.valueOf() as Record<string, unknown>;
+    const props = {
+        name: json.name,
+    } as Record<string, unknown>;
+    if (json.id !== undefined) {
+        props.id = json.id;
+    }
+    for (const key in json) {
+        if (key === "id" || key === "name" || key === "tag") {
+            continue;
+        }
+        props[key] = json[key];
+    }
+    if (this.children.length) {
+        props.children = this.children.length;
+    }
+    if (!inspect) {
+        inspect = (value: unknown) => `${value}`;
+    }
+    return `${inspect(props, options)}`.replace(/^\{/, `${decamelize(this.tag)} {`);
+});
 
 export namespace Model {
     /**

